src/language_model/pipeline.py

```diff
@@ -0,0 +1,76 @@
+import os
+import pathlib
+import sys
+from abc import ABC
+from logging import INFO, FileHandler, Formatter, Logger, StreamHandler, getLogger
+from typing import Sequence
+
+sys.path.append(str(pathlib.Path().absolute()))
+
+LOGGING_FORMAT: str = "%(asctime)s : %(levelname)s : %(module)s : %(message)s"
+DEFAULT_LOG_DIR: str = "logs"
+DEFAULT_LOG_FILE: str = "log.txt"
+DEFAULT_CONFIGURATION_DIR: str = "configs"
+
+
+class ITask(object):
+    def execute(self, environment_path: str) -> None:
+        raise NotImplementedError()
+
+
+class Sandbox(object):
+    def get_sandbox_folder_path(self) -> str:
+        raise NotImplementedError()
+
+
+class AbstractSandbox(Sandbox):
+    def __init__(self, sandbox_folder_path: str) -> None:
+        self.sandbox_folder_path = sandbox_folder_path
+
+    def get_sandbox_folder_path(self) -> str:
+        return self.sandbox_folder_path
+
+
+class SandboxTask(ITask, AbstractSandbox, ABC):
+    def __init__(self, sandbox_folder_path: str = "outputs") -> None:
+        super().__init__(sandbox_folder_path)
+
+
+def identifiers_from_config_file(filepath: str) -> Sequence[str]:
```

Comment:
The method looks ambiguous:
-  What are identifiers? 
- imo it would be better to name it `*_from_config_path`.

Reply:
Each experiment step has a set of identifiers that exhaustively describe it - language, step name, version, etc. We use an ordered `str` sequence to not fixate on some particular identifiers.